models/base.py

```diff
@@ -166,6 +166,10 @@ def __call__(self, *args, name: Optional[str] = None, **kwargs) -> Layer:
       if self.calls:
         name_ctx.is_repeated_call = True
       layer_dict = self.make_layer_dict(*args, **kwargs)
+      ret = None
+      if isinstance(layer_dict, Tuple):
```

Comment:
There should not be such an `isinstance(..., Tuple)` check, but this should be done via `nest` instead.

Reply:
Ah whoops sorry, no this is fine here but the check should be different.